packages/jest-runner/src/configLoaders/ReactScriptsJestConfigLoader.ts

```diff
@@ -13,16 +13,30 @@ export default class ReactScriptsJestConfigLoader implements JestConfigLoader {
   }
 
   public loadConfig(): jest.Configuration {
-    // Get the location of react script, this is later used to generate the Jest configuration used for React projects.
-    const reactScriptsLocation = path.join(this.loader.resolve('react-scripts/package.json'), '..');
+    try {
+      // Get the location of react script, this is later used to generate the Jest configuration used for React projects.
+      const reactScriptsLocation = path.join(this.loader.resolve('react-scripts/package.json'), '..');
 
-    // Create the React configuration for Jest
-    const jestConfiguration = this.createJestConfig(reactScriptsLocation);
+      // Create the React configuration for Jest
+      const jestConfiguration = this.createJestConfig(reactScriptsLocation);
 
-    // Set test environment to jsdom (otherwise Jest won't run)
-    jestConfiguration.testEnvironment = 'jsdom';
+      // Set test environment to jsdom (otherwise Jest won't run)
+      jestConfiguration.testEnvironment = 'jsdom';
 
-    return jestConfiguration;
+      return jestConfiguration;
+    }
+    catch (e) {
+      if (this.isNodeErrnoException(e) && e.code === 'MODULE_NOT_FOUND') {
+        throw Error('Unable to locate package react-scripts. ' +
+          'This package is required when projectType is set to "react".');
+      } else {
```

Comment:
`else` is not necessary here since `throw` works +- like `return`

Reply:
Done